parity-crypto/src/hmac/mod.rs

```diff
@@ -44,19 +46,19 @@ impl<T> Deref for Signature<T> {
 pub struct SigKey<T>(KeyInner, PhantomData<T>);
 
 enum KeyInner {
-	Sha256(GenericArray<u8, U64>),
-	Sha512(GenericArray<u8, U128>),
+	Sha256(Vec<u8>),
```

Comment:
Can you explain why `GenericArray<u8, T>` is replaced by `Vec<u8`?

Reply:
Absolutely, should have been in the PR description. When replacing the HMAC code (used ring) with the `hmac` crate a change was introduced that required calling code to provide a full 128/64-bytes long key for SHA512/SHA256 (stored in the form of a `GenericArray`, sized to the digest block length). 
Using a `Vec` to store the incoming key removes the fixed size requirement and using `new_varkey()` ensures too long/too short keys are handled correctly.